### PR TITLE
Fix exception during setup of grant types of application.

### PR DESCRIPTION
**Pull request by Meruja**
Fix wso2/product-apim/issues/7186

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 1]
```diff
@@ -49,8 +49,22 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
             settingsDTO.setApplicationSharingEnabled(APIUtil.isMultiGroupAppSharingEnabled());
             settingsDTO.setMapExistingAuthApps(APIUtil.isMapExistingAuthAppsEnabled());
             Map<String, Environment> environments = APIUtil.getEnvironments();
-            Environment environment = environments.get(RestApiConstants.DEFAULT_ENVIRONMENT);
-            settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
+            if (environments.size()==1){
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: Please format the code.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 2]
```diff
@@ -49,8 +49,22 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
...
+                Map.Entry<String,Environment> entry = environments.entrySet().iterator().next();
+                Environment singleEnvironment = environments.get(entry.getKey());
+                settingsDTO.apiGatewayEndpoint(singleEnvironment.getApiGatewayEndpoint());
+            } else {
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: I don't think we need if-else here. Else logic can handle single GW scenario too right?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java (hunk 1 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: Fixed with 23318e4

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java (hunk 2 above)
Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: Fixed with abae412

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 3]
```diff
@@ -49,8 +48,16 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
             settingsDTO.setApplicationSharingEnabled(APIUtil.isMultiGroupAppSharingEnabled());
             settingsDTO.setMapExistingAuthApps(APIUtil.isMapExistingAuthAppsEnabled());
             Map<String, Environment> environments = APIUtil.getEnvironments();
-            Environment environment = environments.get(RestApiConstants.DEFAULT_ENVIRONMENT);
-            settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
+            for (Map.Entry<String, Environment> entry : environments.entrySet()) {
+                Environment multipleEnvironment = environments.get(entry.getKey());
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]:
```suggestion
                Environment environment = environments.get(entry.getKey());
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 4]
```diff
@@ -49,8 +48,16 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
...
+                if (multipleEnvironment.isDefault()) {
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]:
```suggestion
                if (environment.isDefault()) {
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 5]
```diff
@@ -49,8 +48,16 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
...
+                    settingsDTO.apiGatewayEndpoint(multipleEnvironment.getApiGatewayEndpoint());
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]:
```suggestion
                    settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 6]
```diff
@@ -49,8 +48,16 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
...
+                    break;
+                }
+            }
+            Map.Entry<String, Environment> entry = environments.entrySet().iterator().next();
+            Environment singleEnvironment = environments.get(entry.getKey());
+            settingsDTO.apiGatewayEndpoint(singleEnvironment.getApiGatewayEndpoint());
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: aren't we replacing the value added at line 54 by this one here?

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java]: Fixed with 1bd989f

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java [hunk 7]
```diff
@@ -49,8 +48,22 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
             settingsDTO.setApplicationSharingEnabled(APIUtil.isMultiGroupAppSharingEnabled());
             settingsDTO.setMapExistingAuthApps(APIUtil.isMapExistingAuthAppsEnabled());
             Map<String, Environment> environments = APIUtil.getEnvironments();
-            Environment environment = environments.get(RestApiConstants.DEFAULT_ENVIRONMENT);
-            settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
+            if (environments.size() == 0) {
+                settingsDTO.apiGatewayEndpoint("http://localhost:8280,https://localhost:8243");
```
Comment (isharac) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java:52]: a space required.

Comment (Meruja) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java:52]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9062

**Pull request merged**